server/src/main/java/org/apache/druid/segment/loading/SegmentLocalCacheManager.java

```diff
@@ -436,6 +453,58 @@ public void cleanup(DataSegment segment)
     }
   }
 
+  @Override
+  public void loadSegmentIntoPageCache(DataSegment segment, ExecutorService exec)
+  {
+    ExecutorService execToUse = exec != null ? exec : loadSegmentsIntoPageCacheOnDownloadExec;
+    if (execToUse == null) {
+      return;
+    }
+
+    execToUse.submit(
+        () -> {
+          final ReferenceCountingLock lock = createOrGetLock(segment);
```

Comment:
@wjhypo can you add test for this runnable, travis is failing because of code coverage issue. Once travis passes will merge this PR.

Reply:
I have fixed some of the docs related spelling check failures